src/vs/workbench/contrib/terminal/browser/terminalStatusList.ts

```diff
@@ -99,7 +99,7 @@ export class TerminalStatusList extends Disposable implements ITerminalStatusLis
 			this._statuses.set(status.id, status);
 			this._onDidAddStatus.fire(status);
 			const newPrimary = this.primary;
-			if (oldPrimary !== newPrimary) {
+			if (oldPrimary !== newPrimary && this.primary?.icon) {
```

Comment:
this is still not right. it fixes the bug, but for some reason, now toggling auto attach does not update the icon

Reply:
I see we have to tweak the `primary` function

Reply:
not that simple bc seems the task status gets removed first

<img width="885" alt="Screenshot 2023-09-20 at 11 23 16 AM" src="https://github.com/microsoft/vscode/assets/29464607/7f8cf3b2-c64b-4d76-8ff1-02349953402b">